backend/sources/google/events-router/src/main/java/co/airy/core/sources/google/EventsRouter.java

```diff
@@ -0,0 +1,124 @@
+package co.airy.core.sources.google;
+
+import co.airy.avro.communication.Channel;
+import co.airy.avro.communication.ChannelConnectionState;
+import co.airy.avro.communication.DeliveryState;
+import co.airy.avro.communication.Message;
+import co.airy.avro.communication.SenderType;
+import co.airy.kafka.schema.application.ApplicationCommunicationChannels;
+import co.airy.kafka.schema.application.ApplicationCommunicationMessages;
+import co.airy.kafka.schema.source.SourceGoogleEvents;
+import co.airy.kafka.streams.KafkaStreamsWrapper;
+import co.airy.log.AiryLoggerFactory;
+import co.airy.uuid.UUIDV5;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.KTable;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.util.Map;
+
+@Component
+public class EventsRouter implements DisposableBean, ApplicationListener<ApplicationReadyEvent> {
+    private static final String appId = "sources.google.EventsRouter";
+    private static final Logger log = AiryLoggerFactory.getLogger(EventsRouter.class);
+
+    private final KafkaStreamsWrapper streams;
+    private final ObjectMapper objectMapper;
+
+    public EventsRouter(KafkaStreamsWrapper streams, ObjectMapper objectMapper) {
+        this.streams = streams;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public void destroy() throws Exception {
+        if (streams != null) {
+            streams.close();
+        }
+    }
+
+    @Override
+    public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
+        startStream();
+    }
+
+    private void startStream() {
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        // Channels table
+        KTable<String, Channel> channelsTable = builder.<String, Channel>stream(new ApplicationCommunicationChannels().name())
+                .groupBy((k, v) -> v.getSourceChannelId())
+                .reduce((aggValue, newValue) -> newValue)
+                .filter((sourceChannelId, channel) -> "google".equalsIgnoreCase(channel.getSource())
```

Comment:
This is perhaps a tiny optimization, but I just realized that we can filter for source before grouping and thereby reduce the store size. But then again the channels topic will always be relatively tiny.

Reply:
Yes! Will do it